migration/src/main/java/kin/sdk/migration/MigrationManager.java

```diff
@@ -135,14 +161,12 @@ private void startMigrationProcess(final IMigrationManagerCallbacks migrationMan
 		}
 	}
 
-	private void burnAndMigrateAccount(final IKinClient newKinClient,
+	private void burnAndMigrateAccount(final IKinClient newKinClient, String publicAddress,
 		final IMigrationManagerCallbacks migrationManagerCallbacks) {
 		KinClientCoreImpl kinClientCore = initKinCore();
-		if (kinClientCore.hasAccount()) {
+		if (kinClientCore.hasAccount() && publicAddress != null && !publicAddress.isEmpty()) {
```

Comment:
consider throwing a dedicated exception if public address wasn't found in KinClient

Reply:
ok